src/notation/internal/masternotation.cpp

```diff
@@ -325,6 +325,18 @@ void MasterNotation::applyOptions(mu::engraving::MasterScore* score, const Score
             nvb->setAutoSizeEnabled(tvb->isAutoSizeEnabled());
         }
 
+        for (Page* page : score->pages()) {
```

Comment:
Question about this: would it also be possible to iterate through `score->parts()`, and for each part remove the non-primary instruments from that part? That might be more performant, and feels slightly more straightforward.

Also, we might want to put this inside `clearMeasures`, because it's closely related to that, in my view; the instruments are namely removed because their corresponding instrument change is being removed inside `clearMeasures`.